std/mem.zig

```diff
@@ -435,6 +438,9 @@ pub fn readInt(bytes: []const u8, comptime T: type, endian: builtin.Endian) T {
 /// Reads a big-endian int of type T from bytes.
 /// bytes.len must be exactly @sizeOf(T).
 pub fn readIntBE(comptime T: type, bytes: []const u8) T {
+    if (comptime T.bit_count < 8) {
+      return @compileError("Only integer types with bitwidth >= 8 are allowed; passed " ++ @typeName(T));
```

Comment:
The docs say "`bytes.len` must be exactly `@sizeOf(T)`" but that is not what this compile error is testing for. I think instead of `@sizeOf(T)` it should be talking about `bit_count` because that handles, e.g. `u24` better, where it might make sense for the `@sizeOf` to be 4 even though the bit_count is evenly divisible by 8.

This function should emit a compile error for all integer types where `T.bit_count % 8 != 0`, because those are the ones where the integer might not have enough bits to represent all the data.

Reply:
but it does work for `T.bit_count % 8 != 0`, as shown with the test assert I added:

```zig
assert(readIntBE(u9, []u8{0x0, 0x32}) == 0x32);
assert(readIntLE(u9, []u8{0x12, 0x0}) == 0x12);
```